### ISSUE TITLE
Плагин для перенаправления чтения кода общего модуля и кода модуля формы из epf

### ISSUE THREAD

**Issue by Pr-Mex** [marmyshev/edt-plugins#11]
Плагин позволит менять код модулей без перезапуска конфигурации.
1. Нужна кнопка - перенаправить код модуля в epf.
2.  Вернуть код модуля из epf.

**Comment by marmyshev** [comment]
Это типа для отладки куска общего модуля в обработке "по-быстрому"?

**Comment by Pr-Mex** [comment]
> Это типа для отладки куска общего модуля в обработке "по-быстрому"?

Да. Пишем код и не перезапускаем сеанс 1С.

**Comment by serjevski** [comment]
> 
> 
> > Это типа для отладки куска общего модуля в обработке "по-быстрому"?
> 
> Да. Пишем код и не перезапускаем сеанс 1С.

Так оно ж и так присохранении внешней обработки в ЕДТ генерит в bin новый EPF.

**Comment by marmyshev** [comment]
> > > Это типа для отладки куска общего модуля в обработке "по-быстрому"?
> > 
> > 
> > Да. Пишем код и не перезапускаем сеанс 1С.
> 
> Так оно ж и так присохранении внешней обработки в ЕДТ генерит в bin новый EPF.

Да, но идея - чтобы это делать для «общего модуля» - чтобы если хочешь отладить процедуру в общем модуле - генерится обработка-шаблон в которую копируется процедура из общего модуля и делается вызов при открытии или по кнопке в обработке 

типа  ПКМ по процедуре в редакторе «Отладка в обработке...» и дальше автоматом всё запускается само

**Comment by serjevski** [comment]
Как по мне - это уже излишество. Почему сразу не писать в обработке, а потом копипастить в общий модуль?

**Comment by lyzhenko** [comment]
> > > > Это типа для отладки куска общего модуля в обработке "по-быстрому"?
> > > 
> > > 
> > > Да. Пишем код и не перезапускаем сеанс 1С.
> > 
> > 
> > Так оно ж и так присохранении внешней обработки в ЕДТ генерит в bin новый EPF.
> 
> Да, но идея - чтобы это делать для «общего модуля» - чтобы если хочешь отладить процедуру в общем модуле - генерится обработка-шаблон в которую копируется процедура из общего модуля и делается вызов при открытии или по кнопке в обработке
> 
> типа ПКМ по процедуре в редакторе «Отладка в обработке...» и дальше автоматом всё запускается само

а если расширить данную идею, "hot reload" разработка без перезапуска сеанса в 1С, меняем код в edt, в запущенном сеансе уже все изменилось. @marmyshev как думаете это возможно реализовать ?

**Comment by marmyshev** [comment]
> а если расширить данную идею, "hot reload" разработка без перезапуска сеанса в 1С, меняем код в edt, в запущенном сеансе уже все изменилось.

Идея-то классная, но тут есть проектное поведение Платформы - например "динамическое обновление" в ИБ - да, можно применить, но при этом в текущем сеансе (который подключен в дебаге) будет работать старый код, нужен рестарт сеанса - а это как раз противоречит Hot reload. 

Тут с внешними обработками - сеанс не рестартуется, но при этом "закрывается обработка"  т.е. выбрасывается из памяти старая версия кода. и открывается новая обработка - это как раз компиляция и загрузка в память нового.

причем есть еще ограничение Платформы - что снаружи процесса 1С (работающего, уже запущенного) нельзя закрыть внешнюю обработку и перекрыть внешнюю обработку.
